src/measure_theory/interval_integral.lean

```diff
@@ -1360,4 +1374,57 @@ theorem integral_deriv_eq_sub (hderiv : ∀ x ∈ interval a b, differentiable_a
   ∫ y in a..b, deriv f y = f b - f a :=
 integral_eq_sub_of_has_deriv_at (λ x hx, (hderiv x hx).has_deriv_at) hcont'
 
+/-!
+### Integration by parts
+-/
+
+lemma integral_deriv_mul_eq_sub' {u v u' v' : ℝ → ℝ} (hab : a ≤ b)
+  (hu : ∀ x ∈ Icc a b, has_deriv_at u (u' x) x)
+  (hv : ∀ x ∈ Icc a b, has_deriv_at v (v' x) x)
+  (hcu' : continuous_on u' (Icc a b)) (hcv' : continuous_on v' (Icc a b)) :
+  ∫ x in a..b, u' x * v x + u x * v' x = u b * v b - u a * v a :=
+begin
+  have hcu : continuous_on u (Icc a b) := λ x hx, continuous_at.continuous_within_at
+      (has_deriv_at.continuous_at (hu x hx)),
+  have hcv : continuous_on v (Icc a b) := λ x hx, continuous_at.continuous_within_at
+      (has_deriv_at.continuous_at (hv x hx)),
+  rw integral_eq_sub_of_has_deriv_at,
+  intros x hx;
+  simp only [interval_of_le, hab, min_eq_left, max_eq_right] at hx,
+  { exact (hu x hx).mul (hv x hx) },
+  { simp only [interval_of_le, hab, min_eq_left, max_eq_right],
+    exact (hcu'.mul hcv).add (hcu.mul hcv') }
+end
+
+lemma integral_deriv_mul_eq_sub {u v u' v' : ℝ → ℝ}
+  (hu : ∀ x ∈ interval a b, has_deriv_at u (u' x) x)
+  (hv : ∀ x ∈ interval a b, has_deriv_at v (v' x) x)
+  (hcu : continuous_on u' (interval a b)) (hcv : continuous_on v' (interval a b)) :
+  ∫ x in a..b, u' x * v x + u x * v' x = u b * v b - u a * v a :=
+begin
+  cases le_total a b with hab hab,
+  { simp only [hab, and_imp, max_eq_right, min_eq_left, interval_of_le]
+      at hu hv hcu hcv ⊢,
```

Comment:
Here (and in a couple of other places) you only need `simp only [interval_of_le hab] ...`. This also fits on one line.